packages/ui/src/components/Preview.tsx

```diff
@@ -62,54 +62,50 @@ const Preview = ({ template, inputs, size, onChangeInput }: PreviewReactProps) =
   );
 
   return (
-    <Root ref={rootRef} size={size} scale={scale}>
+    <Root size={size} scale={scale}>
       <CtlBar
-        size={{ height: Math.max(size.height, pageSizesHeightSum), width: size.width }}
+        size={size}
         pageCursor={pageCursor}
         pageNum={schemasList.length}
         setPageCursor={(p) => {
-          if (!rootRef.current) return;
-          rootRef.current.scrollTop = getPagesScrollTopByIndex(pageSizes, p, scale);
+          if (!containerRef.current) return;
+          containerRef.current.scrollTop = getPagesScrollTopByIndex(pageSizes, p, scale);
           setPageCursor(p);
         }}
         zoomLevel={zoomLevel}
-        setZoomLevel={(zoom) => {
-          if (rootRef.current) {
-            rootRef.current.scrollTop = getPagesScrollTopByIndex(pageSizes, pageCursor, scale);
-          }
-          setZoomLevel(zoom);
-        }}
+        setZoomLevel={setZoomLevel}
       />
       <UnitPager
-        size={{ height: Math.max(size.height, pageSizesHeightSum), width: size.width }}
+        size={size}
         unitCursor={unitCursor}
         unitNum={inputs.length}
         setUnitCursor={setUnitCursor}
       />
-
-      <Paper
-        paperRefs={paperRefs}
-        scale={scale}
-        size={size}
-        schemasList={schemasList}
-        pageSizes={pageSizes}
-        backgrounds={backgrounds}
-        renderSchema={({ schema, index }) => {
-          const { key } = schema;
-          const data = (input && input[key]) || '';
-          return (
-            <Renderer
-              key={schema.id}
-              schema={Object.assign(schema, { data })}
-              mode={isForm ? 'form' : 'viewer'}
-              placeholder={template.sampledata?.[0]?.[key] ?? ''}
-              tabIndex={index + 100}
-              onChange={(value) => handleChangeInput({ key, value })}
-              outline={isForm ? '1px dashed #4af' : 'transparent'}
-            />
-          );
-        }}
-      />
+      <div ref={containerRef} style={{ ...size, position: 'relative', overflow: 'auto' }}>
```

Comment:
I have changed Preview to be consistent with the Designer, with a wrapping Div which is viewport/canvas for scrolling.